internal/exec/util/file.go

```diff
@@ -112,6 +114,33 @@ func RenderFile(l *log.Logger, c *resource.HttpClient, f types.File) *File {
 		return nil
 	}
 
+	if f.User.Name != "" {
+		u, err := user.Lookup(f.User.Name)
```

Comment:
Don't we need to lookup the user in a chroot anyways?

If you look at how `passwd` works, it creates a user via the `--root=/sysroot`.

It also has the cgo helper function `userLookup` to similarly check in sysroot.

I'd expect that this, as implemented, would be unable to add a user and then reference it for filesystem ownership.

The easiest fix to that would probably be reusing the cgo function here too.